File: dotnet/Multiplayer/Matchmaking/MatchmakingStartResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking
{
    /// <summary>
    /// Response returned when a game is successfully started from matchmaking.
    /// Contains the created room ID and transfer information.
    /// </summary>
    public class MatchmakingStartResponse : ApiResponse<EMatchmakingStartError>
    {
        /// <summary>
        /// The ID of the created game room.
        /// </summary>
        public string Room_id { get; set; } = string.Empty;

        /// <summary>
        /// The name of the created game room.
        /// </summary>
        public string Room_name { get; set; } = string.Empty;

        /// <summary>
        /// The number of players transferred from matchmaking to the room.
        /// </summary>
        public int Players_transferred { get; set; }

        /// <summary>
        /// Confirmation message for the game start.
        /// </summary>
        public string Message { get; set; } = string.Empty;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingStartResponse.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a game is successfully started from matchmaking.
 * Contains the created room ID and transfer information.
 */
public class MatchmakingStartResponse extends ApiResponse {
    @JsonProperty("room_id")
    private String roomId;

    @JsonProperty("room_name")
    private String roomName;

    @JsonProperty("players_transferred")
    private int playersTransferred;

    @JsonProperty("message")
    private String message;

    public String getRoomId() {
        return roomId;
    }

    public void setRoomId(String roomId) {
        this.roomId = roomId;
    }

    public String getRoomName() {
        return roomName;
    }

    public void setRoomName(String roomName) {
        this.roomName = roomName;
    }

    public int getPlayersTransferred() {
        return playersTransferred;
    }

    public void setPlayersTransferred(int playersTransferred) {
        this.playersTransferred = playersTransferred;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
